install_config/persistent_storage/persistent_storage_nfs.adoc

```diff
@@ -99,66 +93,207 @@ pv0001                   <none>    5368709120   RWO           Available
 
-By default, persistent volumes are set to *Retain*. NFS volumes which are set to
-*Recycle* are scrubbed (i.e., `rm -rf` is run on the volume) after being
-released from their claim (i.e, after the user's `*PersistentVolumeClaim*` bound
-to the volume is deleted). Once recycled, the NFS volume can be bound to a new
-claim.
+As an example, if the target NFS directory appears on the NFS server as:
 
-[[nfs-automation]]
+[[nfs-export]]
+====
+----
+# ls -lZ /opt/nfs -d
+drwxrws---. nobody 5555 unconfined_u:object_r:usr_t:s0   /opt/nfs
 
-== Automation
-As discussed, clusters can be provisioned with persistent storage using NFS in
-the following way:
+# id nobody
+uid=99(nobody) gid=99(nobody) groups=99(nobody)
+----
+====
 
-- Disk partitions can be used to link:#nfs-enforcing-disk-quotas[enforce storage
-quotas].
-- Security can be enforced by link:#nfs-volume-security[restricting volumes] to
-the namespace that has a claim to them.
-- link:#nfs-reclaiming-resources[Reclamation of discarded resources] can be
-configured for each persistent volume.
+Then the container must match SELinux labels, and either run with a UID of 99
+(*nobody* owner) or with 5555 in its supplemental groups in order to access the
+directory.
 
-They are many ways that you can use scripts to automate the above tasks. You can
-use an
-link:https://github.com/openshift/openshift-ansible/tree/master/roles/kube_nfs_volumes[example
-Ansible playbook] to help you get started.
+[NOTE]
+====
+The owner ID of 99 is used as an example. Even though NFS's *root_squash* maps
+*root* (0) to *nobody* (99), NFS exports can have arbitrary owner IDs. Owner 99
+is not required for NFS exports.
+====
+
+[[nfs-supplemental-groups]]
+=== Group IDs
+
+The recommended way to handle NFS access (assuming it is not an option to change
+permissions on the NFS export) is to use supplemental groups. Supplemental
+groups in OpenShift are used for shared storage, of which NFS is an example. In
+contrast, block storage, such as Ceph RBD or iSCSI, use the *fsGroup* SCC
+strategy and  the *fsGroup* value in the pod's `*securityContext*`.
+
+[NOTE]
+====
+It is generally preferable to use supplemental group IDs to gain access to
+persistent storage versus using link:#nfs-user-ids[user IDs]. Supplemental
+groups are covered further in
+link:pod_security_context.html#supplemental-groups[Volume Security].
+====
 
-[[selinux-and-nfs-export-settings]]
+Because the group ID on the link:#nfs-export[example target NFS directory] shown
+above is 5555, the pod can define that group ID using `*supplementalGroups*`
+under the pod's global `*securityContext*` definition. For example:
+
+====
+----
+spec:
+  containers:
+    - name:
+    ...
+      #runAsUser: 99 from above has been commented out here
+  securityContext: <1>
+    supplementalGroups: [5555] <2>
+----
+<1> `*securityContext*` must be defined globally to the pod, not under a
+specific container.
+<2> An array of GIDs defined globally for the pod.
```

Comment:
Suggest being explicit for callout 2, adding something along the lines of "in this case, there is one element in the array -- more GIDs would be comma-separated", for the benefit of readers unfamiliar w/ the inline array YAML syntax.